### PR TITLE
dev_Petrov_03

### PR DESCRIPTION
**Pull request by grom4er**

### CODE REVIEW

**Review by Resci — APPROVED** [review]

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -17,6 +18,16 @@ public UserDaoImpl(SessionFactory sessionFactory) {
         this.sessionFactory = sessionFactory;
     }
 
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            User user = session.get(User.class, id);
+            return Optional.ofNullable(user);
```
Comment (dmytro-hryhoruk) [inline src/main/java/dao/impl/UserDaoImpl.java:25]: let's do this inline

File: src/main/java/dto/UserResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/dto/UserResponseDto.java:16]:
```suggestion
```

File: src/main/java/service/UserMapper.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import dto.UserResponseDto;
+import model.User;
+
+public interface UserMapper {
+    UserResponseDto toDto(User user);
+
+    User fromDto(UserResponseDto userResponseDto);
+}
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/service/UserMapper.java:11]:
```suggestion
```

File: src/main/java/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -19,6 +19,14 @@ public void add(User user) {
         userDao.add(user);
     }
 
+    @Override
+    public User getUserById(Long id) {
+        if (userDao.getById(id).isEmpty()) {
+            return null;
+        }
+        return userDao.getById(id).get();
```
Comment (dmytro-hryhoruk) [inline src/main/java/service/impl/UserServiceImpl.java:27]: you are requesting db twice

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @GetMapping("/{user_id}")
+    public UserResponseDto getUser(@PathVariable("user_id") Long id) {
+        return userMapper.toDto(userService.getUserById(id));
+    }
+
+    @GetMapping("/getAll")
```
Comment (dmytro-hryhoruk) [inline src/main/java/controller/UserController.java:31]:
```suggestion
    @GetMapping
```

File: src/main/java/service/impl/UserMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        UserResponseDto userDto = new UserResponseDto();
+        userDto.setName(user.getName());
+        return userDto;
+    }
+
+    public User fromDto(UserResponseDto inputUserDto) {
+        User user = new User();
+        user.setName(inputUserDto.getName());
+        return user;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/service/impl/UserMapperImpl.java:20]: we don't need this method now

File: src/main/java/dto/UserResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package dto;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserResponseDto {
+    private String name;
```
Comment (dmytro-hryhoruk) [inline src/main/java/dto/UserResponseDto.java:7]: let's add id too

File: src/main/java/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -19,6 +20,15 @@ public void add(User user) {
         userDao.add(user);
     }
 
+    @Override
+    public User getUserById(Long id) {
+        Optional<User> user = userDao.getById(id);
+        if (user.isEmpty()) {
+            return null;
+        }
+        return user.get();
```
Comment (dmytro-hryhoruk) [inline src/main/java/service/impl/UserServiceImpl.java:29]: return userDao.getById(id).orElse(null);

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (dmytro-hryhoruk) [reply src/main/java/service/impl/UserServiceImpl.java:29]: but let's use orElseThrow() instead; It's better to throw exceptions in such cases. Later we will learn how our controller can handle such situations

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**